ui/app/adapters/kubernetes/role.js

```diff
@@ -32,7 +32,7 @@ export default class KubernetesRoleAdapter extends NamedPathAdapter {
   }
   generateCredentials(backend, data) {
     const generateCredentialsUrl = `${this.buildURL()}/${encodePath(backend)}/creds/${data.role}`;
-
+    delete data.role;
```

Comment:
Could this also be added to the serializer as `serialize: false` ?

Reply:
Since this is a custom adapter method the serializer is not hit. We decided not to have a dedicated model for role credentials since they are not persisted in the app which is why we are using the role adapter for the request.

Reply:
(not suggesting, just curious)